src/libstd/sys/unix/os.rs

```diff
@@ -206,7 +206,7 @@ pub fn current_exe() -> io::Result<PathBuf> {
         if err != 0 { return Err(io::Error::last_os_error()); }
         if sz == 0 { return Err(io::Error::last_os_error()); }
         v.set_len(sz as uint - 1); // chop off trailing NUL
-        Ok(PathBuf::new::<OsString>(&OsStringExt::from_vec(v)))
+        Ok(PathBuf::new::<OsString>(OsStringExt::from_vec(v)))
```

Comment:
For future reference (I only discovered this recently!) this can also be replaced by `PathBuf::new(OsString::from_vec(v))` which removes the need for type ascription :)